### PR TITLE
[Port Hil-SERL] Add unit tests for the reward classifier & fix imports & check script

### PR DESCRIPTION
**Pull request by helper2424**
## What this does
Add unit tests and check script for the Reward Classifier. Also, fixed small issue with imports.

## How it was tested
- I have added unit tests for the reward classifier `/tests/policies/hilserl/classifier/test_modelling_classifier.py`. Run `python -m pytest -sv ./tests/policies/hilserl/classifier/test_modelling_classifier.py `
- Also I have added a script that trains and check metrics for the classifier, check - `python tests/policies/hilserl/classifier/check_hiserl_reward_classifier.py`. If u want to see a convergence for the model - choose bigger number of epoch in the file, check `EPOCH_NUM`, setup 30-40.

## How to checkout & try? (for the reviewer)
Provide a simple way for the reviewer to try out your changes.

Examples:
```bash
python -m pytest -sv ./tests/policies/hilserl/classifier/test_modelling_classifier.py 
python tests/policies/hilserl/classifier/check_hiserl_reward_classifier.py
```

### CODE REVIEW

File: lerobot/scripts/check_hiserl_reward_classifier.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+    logging.info(f'Accuracy: {acc:.2f}')
+    logging.info(f'Precision: {prec:.2f}')
+    logging.info(f'Recall: {rec:.2f}')
+    logging.info(f'F1 Score: {f1_score:.2f}')
+    logging.info(f'AUROC Score: {auroc_score:.2f}')
+
+if __name__ == "__main__":
+    train_evaluate_multiclass_classifier()
+    train_evaluate_binary_classifier()
```
Comment (helper2424) [inline lerobot/scripts/check_hiserl_reward_classifier.py:222]:
```suggestion
    train_evaluate_binary_classifier()

```

File: lerobot/common/policies/hilserl/classifier/configuration_classifier.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ class ClassifierConfig:
     hidden_dim: int = 256
     dropout_rate: float = 0.1
     model_name: str = "microsoft/resnet-50"
-    device: str = "cuda" if torch.cuda.is_available() else "mps"
+    device: str = "cpu"
```
Comment (helper2424) [inline lerobot/common/policies/hilserl/classifier/configuration_classifier.py:16]: Let's use `CPU` as default, the policty should apply required device via the config

File: lerobot/common/policies/hilserl/classifier/modeling_classifier.py [hunk 3]
```diff
@@ -69,6 +74,8 @@ def _setup_cnn_backbone(self):
             self.feature_dim = self.encoder.config.hidden_sizes[-1]  # Last channel dimension
         else:
             raise ValueError("Unsupported CNN architecture")
+        
+        self.encoder = self.encoder.to(self.config.device)
```
Comment (helper2424) [inline lerobot/common/policies/hilserl/classifier/modeling_classifier.py:78]: Explicitly apply device

File: pyproject.toml [hunk 4]
```diff
@@ -71,6 +71,9 @@ pyrender = {git = "https://github.com/mmatl/pyrender.git", markers = "sys_platfo
 hello-robot-stretch-body = {version = ">=0.7.27", markers = "sys_platform == 'linux'", optional = true}
 pyserial = {version = ">=3.5", optional = true}
 jsonlines = ">=4.0.0"
+transformers = "^4.47"
```
Comment (helper2424) [inline pyproject.toml:74]: Added transformers as we use it in the classifier, but I didn't see it in deps. Without it the library crashes. Probably nobody noticed that issue as u have it installed globaly.

File: pyproject.toml [hunk 5]
```diff
@@ -71,6 +71,9 @@ pyrender = {git = "https://github.com/mmatl/pyrender.git", markers = "sys_platfo
...
+torchmetrics = "^1.6.0"
```
Comment (helper2424) [inline pyproject.toml:75]: Added to track default models metrices, I have used them to check the reward classifier.

File: tests/policies/hilserl/classifier/check_hiserl_reward_classifier.py [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+    DEVICE = torch.device("mps")
+else:
+    DEVICE = torch.device("cpu")
+
+
+def train_evaluate_multiclass_classifier():
+    logging.info(
+        f"Start multiclass classifier train eval with {DEVICE} device, batch size {BATCH_SIZE}, learning rate {LR}"
+    )
+    multiclass_config = ClassifierConfig(model_name="microsoft/resnet-50", device=DEVICE, num_classes=10)
```
Comment (ChorntonYoel) [inline tests/policies/hilserl/classifier/check_hiserl_reward_classifier.py:47]:
```suggestion
    multiclass_config = ClassifierConfig(model_name="microsoft/resnet-18", device=DEVICE, num_classes=10)
```
It's a bit slow with the resnet-50 imo.

File: tests/policies/hilserl/classifier/check_hiserl_reward_classifier.py [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+from torch.optim import Adam
+from torch.utils.data import DataLoader
+from torchmetrics import AUROC, Accuracy, F1Score, Precision, Recall
+from torchvision.datasets import CIFAR10
+from torchvision.transforms import ToTensor
+
+from lerobot.common.policies.hilserl.classifier.modeling_classifier import Classifier, ClassifierConfig
+
+BATCH_SIZE = 1000
+LR = 0.1
```
Comment (ChorntonYoel) [inline tests/policies/hilserl/classifier/check_hiserl_reward_classifier.py:32]: I struggled to get signal with those hyperparameters in terms of convergence/non-random acc. I divided bsz by 10 and lr by 1000. Nothing optimal, but I got 43% acc vs 10% and the loss was clearly converging, so that's good enough to get signal

**Review by ChorntonYoel — APPROVED** [review]
Thanks for the thorough tests on the classifier! lgtm
I'll let @michel-aractingi chime in for the new dependencies
